packages/devtools_app/lib/src/memory/flutter/memory_controller.dart

```diff
@@ -597,35 +1017,130 @@ class MemoryTimeline {
+
+    if (endingIndex - startingIndex != liveData.length || reloadAllData) {
+      // Process the data received (startingDataIndex is the last sample).
+      final args = _processData(endingIndex < 0 ? 0 : endingIndex);
+
+      // Debugging data - to enable remove logical not operator.
+      if (!true) {
+        final DateFormat mFormat = DateFormat('hh:mm:ss.mmm');
+        final startDT = mFormat.format(DateTime.fromMillisecondsSinceEpoch(
+            liveData[startingIndex].timestamp.toInt()));
+        final endDT = mFormat.format(DateTime.fromMillisecondsSinceEpoch(
+            liveData[endingIndex].timestamp.toInt()));
+        print('Time range Live data start: $startDT, end: $endDT');
+      }
+
+      return args;
     }
 
     return [];
   }
 
+  List<Map> recomputeOfflineData(int displayInterval) {
+    assert(displayInterval > 0);
+
+    // Compute a new starting index from length - N minutes.
+    final timeLastSample = data.last.timestamp;
+    var dataIndex = data.length - 1;
+    for (; dataIndex > 0; dataIndex--) {
+      final sample = data[dataIndex];
+      final timestamp = sample.timestamp;
+
+      if ((timeLastSample - timestamp) > displayInterval) break;
+    }
+
+    startingIndex = dataIndex;
+
+    // Debugging data - to enable remove logical not operator.
+    if (!true) {
+      final DateFormat mFormat = DateFormat('hh:mm:ss.mmm');
+      final startDT = mFormat.format(DateTime.fromMillisecondsSinceEpoch(
+          data[startingIndex].timestamp.toInt()));
+      final endDT = mFormat.format(DateTime.fromMillisecondsSinceEpoch(
+          data[endingIndex].timestamp.toInt()));
+      print('Recompute Time range Offline data start: $startDT, end: $endDT');
+    }
+
+    // Start from the first sample to display in this time interval.
+    return _processData(startingIndex);
+  }
+
+  List<Map> recomputeLiveData(int displayInterval) {
+    assert(displayInterval > 0);
+
+    // Compute a new starting index from length - N minutes.
+    final timeLastSample = data.last.timestamp;
+    var dataIndex = data.length - 1;
+    for (; dataIndex > 0; dataIndex--) {
+      final sample = data[dataIndex];
+      final timestamp = sample.timestamp;
+
+      if ((timeLastSample - timestamp) > displayInterval) break;
+    }
+
+    startingIndex = dataIndex;
+
+    // Debugging data - to enable remove logical not operator.
+    if (!true) {
+      final DateFormat mFormat = DateFormat('hh:mm:ss.mmm');
+      final startDT = mFormat.format(DateTime.fromMillisecondsSinceEpoch(
+          data[startingIndex].timestamp.toInt()));
+      final endDT = mFormat.format(DateTime.fromMillisecondsSinceEpoch(
+          data[endingIndex].timestamp.toInt()));
+      print('Recompute Time range Live data start: $startDT, end: $endDT');
+    }
+
+    // Start from the first sample to display in this time interval.
+    return _processData(startingIndex);
+  }
+
+  static String jsonPayloadField = 'samples';
```

Comment:
put static members at top of class

Reply:
make them private

  static const String _jsonPayloadField = 'samples';
  static const String _jsonVersionField = 'version';
  static const String _jsonDataField = 'data';

and kept them near the methods where they are used.  Not for external access.